File: src/com/zypex/piflow/profile/ProfileSegment.java
package com.zypex.piflow.profile;

import utils.math.BoundedFunction;
import utils.math.Function;
import utils.math.Vector;

public abstract class ProfileSegment implements BoundedFunction<Derivatives<Vector>> {

    public abstract double getLength();

    public double getT(double x, double y){
        return getT(new Vector(x,y));
    }

    @Override
    public abstract ProfileSegment offset(double offset) throws ClassCastException;

    public abstract double getT(Vector pos);

}

File: src/com/zypex/piflow/profile/ProfileSegment.kt
package com.zypex.piflow.profile

import com.zypex.piflow.math.BoundedFunction
import com.zypex.piflow.math.Vector

abstract class ProfileSegment : BoundedFunction<Derivatives<Vector>> {
    abstract val length: Double
    fun getT(x: Double, y: Double): Double {
        return getT(Vector(x, y))
    }

    abstract override fun offset(offset: Double): ProfileSegment
    abstract fun getT(pos: Vector): Double
}